src/Options/Generic.hs

```diff
@@ -826,18 +859,68 @@ getRecordPure
     -- ^ Command-line arguments
     -> Maybe a
 getRecordPure args = do
-    let prefs = Options.ParserPrefs
-            { prefMultiSuffix     = "..."
-            , prefDisambiguate    = False
-            , prefShowHelpOnError = False
-            , prefBacktrack       = True
-            , prefColumns         = 80
-#if MIN_VERSION_optparse_applicative(0,13,0)
-            , prefShowHelpOnEmpty = False
-#else
-#endif
-            }
     let header = Options.header ""
     let info   = Options.info parseRecord header
     let args'  = map Data.Text.unpack args
-    Options.getParseResult (Options.execParserPure prefs info args')
+    Options.getParseResult (Options.execParserPure ogPrefs info args')
+
+-- | @optparse-generic@'s flavor of options
+ogPrefs :: Options.ParserPrefs
+ogPrefs = Options.defaultPrefs
+  { Options.prefMultiSuffix = "..."
+  }
+
+-- | A type family to extract fields wrapped using '(<?>)'
+type family (:::) (wrap :: Wrap) wrapped
+type instance 'Wrap ::: wrapped = wrapped
+type instance 'Unwrap ::: (field <?> helper) = field
+
+infixr 0 :::
+
+-- | Type level switch for '(:::)'
+data Wrap = Wrap | Unwrap
```

Comment:
Sure! Omitting the apostrophe results in warnings though, since they help keep namespaces for types and values separate. 

I'm going to simply declare `Unwrapped` and `Wrapped` as two empty types. I was not really leveraging DataKinds anyway. It clarified some intent but it had little practical benefits.